include/swift/AST/TypeCheckRequests.h

```diff
@@ -3128,6 +3128,24 @@ class CheckRedeclarationRequest
                            evaluator::SideEffect) const;
 };
 
+/// Resolve a given custom attribute to an attached macro declaration.
+class ResolveAttachedMacroRequest
+    : public SimpleRequest<ResolveAttachedMacroRequest,
+                           MacroDecl *(CustomAttr *, DeclContext *),
```

Comment:
Should this return a `ConcreteDeclRef` that points at the `MacroDecl`, so we also have the inferred generic arguments? Otherwise, they won't be captured anywhere.